docs/reference/encryption-and-security.md

```diff
@@ -0,0 +1,154 @@
+4. **Configures Default Keys**: Uses AWS default KMS keys (`kmsKeyId: '*'`) for encryption operations
+
+### Generated Infrastructure
+
+The framework generates the following serverless configuration:
+
+```yaml
+# IAM Permissions
+provider:
+  iamRoleStatements:
+    - Effect: Allow
+      Action:
+        - kms:GenerateDataKey
+        - kms:Decrypt
+      Resource: 
+        - '${self:custom.kmsGrants.kmsKeyId}'
+
+# Environment Variables
+provider:
+  environment:
+    KMS_KEY_ARN: '${self:custom.kmsGrants.kmsKeyId}'
+
+# Plugins
+plugins:
+  - serverless-kms-grants
+
+# Custom Configuration
+custom:
+  kmsGrants:
+    kmsKeyId: '*'
+```
+
+## Using KMS in Your Code
+
+### Accessing the KMS Key ARN
+
+The KMS key ARN is available in your Lambda functions via environment variables:
+
+```javascript
+const kmsKeyArn = process.env.KMS_KEY_ARN;
+
+// Use with AWS SDK for encryption operations
+const { KMSClient, GenerateDataKeyCommand, DecryptCommand } = require('@aws-sdk/client-kms');
+
+const kmsClient = new KMSClient({ region: 'us-east-1' });
+```
+
+### Integration with Frigg Encrypt Module
+
+If you're using the `@friggframework/encrypt` module, it will automatically use the configured KMS key:
+
+```javascript
+const { encrypt, decrypt } = require('@friggframework/encrypt');
+
+// Encrypt sensitive data
+const encryptedData = await encrypt(sensitiveString);
+
+// Decrypt when needed
+const decryptedData = await decrypt(encryptedData);
+```
+
+## Security Best Practices
+
+### When to Use KMS
+
+Enable KMS encryption when your integrations handle:
+
+- Personal Identifiable Information (PII)
+- Financial data
+- Authentication tokens (beyond basic OAuth)
+- Sensitive business data
+- Healthcare information (PHI)
+
+### Key Management
+
+- **Default Keys**: Frigg uses AWS default KMS keys (`*`) for simplicity
+- **Custom Keys**: For enhanced security, consider creating dedicated KMS keys per environment
+- **Key Rotation**: AWS automatically rotates default keys annually
+
+## Deployment Considerations
+
+### Prerequisites
+
+Ensure your deployment environment has:
+
+1. **IAM Permissions**: Deployment role needs KMS permissions to create grants
+2. **KMS Access**: Lambda execution role will have KMS permissions after deployment
+
+### Environment Isolation
+
+KMS configurations are environment-specific:
+
+- **Development**: Uses same default keys for testing
+- **Staging**: Can use environment-specific keys
+- **Production**: Should use dedicated production keys for maximum security
+
+### Version Requirements
+
+- **Framework Version**: Requires `@friggframework/devtools` v2.1.0+
+- **AWS Provider**: Compatible with all AWS regions
+- **Node.js**: Works with all supported Node.js versions (16.x, 18.x, 20.x)
+
+## Examples
+
+### Basic Setup
+
+```javascript
+// app-definition.js
+const appDefinition = {
+    name: 'secure-integration-app',
+    integrations: [
+        require('@friggframework/salesforce'),
```

Comment:
These should mimic pointing to a local integration file instead of the API module